internal/bazel_integration_test/test_runner.js

```diff
@@ -0,0 +1,250 @@
+  fs.writeFileSync(bazelrcFile, bazelrcContents);
+  if (DEBUG)
+    console.log(`\n\n.bazelrc file after appending:\n========================================================================================\n${
+        bazelrcContents}\n^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n`);
+}
+
+// Handle WORKSPACE replacements
+{
+  const workspaceFile = path.posix.join(workspaceRoot, 'WORKSPACE');
+  let workspaceContents = fs.readFileSync(workspaceFile, {encoding: 'utf-8'});
+  // replace repositories
+  for (const repositoryKey of Object.keys(config.repositories)) {
+    const archiveFile = require.resolve(config.repositories[repositoryKey]);
+    const regex =
+        new RegExp(`(local_repository|http_archive|git_repository)\\(\\s*name\\s*\\=\\s*"${
+            repositoryKey}"[^)]+`);
+    const replacement =
+        `load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")\nhttp_archive(\n  name = "${
+            repositoryKey}",\n  url="file:${archiveFile.replace(/\\/g, '/')}"\n`;
+    workspaceContents = workspaceContents.replace(regex, replacement);
+  }
+  fs.writeFileSync(workspaceFile, workspaceContents);
+  if (DEBUG)
+    console.log(`\n\nWORKSPACE file with replacements:\n========================================================================================\n${
+        workspaceContents}\n^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n`);
+}
+
+// Handle package.json replacements
+const packageJsonFile = path.posix.join(workspaceRoot, 'package.json');
+if (isFile(packageJsonFile)) {
+  let packageJsonContents = fs.readFileSync(packageJsonFile, {encoding: 'utf-8'});
+
+
+  const npmPackageKeys = Object.keys(config.npmPackages);
+  if (npmPackageKeys.length) {
+    for (const packageJsonKey of npmPackageKeys) {
+      if (DEBUG) console.log(`\n\ncopying npm package ${packageJsonKey} to tmp`);
+      const packagePath = copyNpmPackage(config.npmPackages[packageJsonKey]);
+      const regex = new RegExp(`\"${packageJsonKey}\"\\s*\:\\s*\"[^"]+`)
+      const replacement = `"${packageJsonKey}": "file:${packagePath.replace(/\\/g, '/')}`;
+      packageJsonContents = packageJsonContents.replace(regex, replacement);
+    }
+    fs.writeFileSync(packageJsonFile, packageJsonContents);
+  }
+
+  const packageJsonReplacementKeys = Object.keys(config.packageJsonRepacements);
+  if (packageJsonReplacementKeys.length) {
+    for (const packageJsonKey of packageJsonReplacementKeys) {
+      const regex = new RegExp(`\"${packageJsonKey}\"\\s*\:\\s*\"[^"]+`)
+      const replacement = `"${packageJsonKey}": "${config.packageJsonRepacements[packageJsonKey]}`;
+      packageJsonContents = packageJsonContents.replace(regex, replacement);
+    }
+  }
+
+  if (DEBUG)
+    console.log(`\n\npackage.json file with replacements:\n========================================================================================\n${
```

Comment:
if you really want the log formatted with these section breaks, maybe a `debug()` helper function would be better

Reply:
👍